### DISCUSSION TITLE
[client] - Client can not detect the agents

### DISCUSSION THREAD

**Discussion by 9to1url** [narrowlink/narrowlink#11] [Q&A] [answered]
**Describe the bug**
I followed the basic tutorial, configured gateway/agent/client. on client when run this:
```
❯ narrowlink list
Agent not found
```

**Additional context**
I am not sure when generate the client token do I need to fill in the agent token in token-generator.yaml ?

**Comment by 9to1url** [comment]
Also could you advice where is the best place to chat?
I joined your Telegram channel, and also followed your Twitter :-)

**Reply by SajjadPourali** [reply]
We can continue here if you wish. : [discussions](https://github.com/narrowlink/narrowlink/discussions)

**Reply by 9to1url** [reply]
sure :-)

**Comment by 9to1url** [comment]
this is message from gateway:
```
[2023-08-16T01:34:46Z WARN  rustls::msgs::handshake] Illegal SNI hostname received "11.22.33.44"
```

**Reply by SajjadPourali** [reply]
Could you please confirm whether the gateway could issue a certificate for you?
You can verify this by checking the certificate folder or by directly opening the gateway address using the HTTPS protocol, like `https://domain.example/`.

**Reply by SajjadPourali** [reply]
You only need to connect to the gateway using the domain name if you are using the `Wss` protocol. This warning is not very important and indicates that the gateway is intended to be accessed using an IP address.

**Reply by 9to1url** [reply]
the page shows, the certificate has a lock and verified is secure:
404 Not Found
The requested resource could not be found.

Powered By [Narrowlink](https://narrowlink.com/)

**Reply by SajjadPourali** [reply]
It is fine, because you didn't setup default host.

**Reply by 9to1url** [reply]
how I continue? should I add the Agent token in Client and regenerate the client token again?

**Reply by SajjadPourali** [reply]
What do you get after running the agent command?
Something like the following output is expected:
```
[2023-08-16T02:34:14Z INFO  narrowlink_agent] Connecting to gateway: domain:443
[2023-08-16T02:34:14Z INFO  narrowlink_agent] Connection successful
```

**Reply by 9to1url** [reply]
$ narrowlink-agent
[2023-08-16T01:45:16Z INFO  narrowlink_agent] Connecting to gateway: gateway.domain.info:443
[2023-08-16T01:45:17Z INFO  narrowlink_agent] Connection successful

**Reply by SajjadPourali** [reply]
It is not important to put both tokens into the same file or create them individually. The same `uid` must be choose in both clients and agents.

**Reply by SajjadPourali** [reply, marked as answer]
Did you choose the same `uid` for both client and agent tokens?

**Reply by 9to1url** [reply]
no, I used uuid v4 to generate them separately. let me change that in client. and should the client name be the same?

**Reply by SajjadPourali** [reply]
If you run the gateway by setting `RUST_LOG=narrowlink_gateway=trace` environment variables, you can view more verbose information in the gateway.

like:
```
[2023-08-16T02:37:48Z DEBUG narrowlink_gateway::state] Agent a825099b-c542-4734-85ea-8b13c73636a1:miranda (11.22.33.44:33210) connected
[2023-08-16T02:37:55Z DEBUG narrowlink_gateway::state] Client a825099b-c542-4734-85ea-8b13c73636a1:c7a94e8e-fd4c-48a1-a53e-7a598b0d1bd5 connected
```

Could you please confirm that both the client and agents that have connected share the same `uid`, and you can see a message similar to the one above?

**Reply by 9to1url** [reply]
after I used same uuid, it working great :-)
thank you very much.

❯ narrowlink list
serveragent:
	Address: 22.33.44.55:43982
	Connection Ping: 41ms

**Reply by SajjadPourali** [reply]
The name is not significant here, but all the agents and clients within one user's space must share the same uid.

**Reply by 9to1url** [reply]
after setup this, let me continue to read the extension docs, I want to access  a web server run on agent machine.
currently I setup my local machine as client, one VPS as gateway and another VPS as agent.
